eva-accession-clustering-automation/clustering_automation/cluster_from_mongo.py

```diff
@@ -0,0 +1,126 @@
+    results = get_all_results_for_query(metadata_connection_handle, query)
+    if len(results) == 0:
+        raise Exception("Could not find assemblies pertaining to taxonomy ID: " + taxonomy_id)
+    return [result[0] for result in results], results[0][1]
+
+
+def get_common_clustering_properties(common_clustering_properties_file):
+    return yaml.load(open(common_clustering_properties_file), Loader=yaml.FullLoader)
+
+
+def generate_linear_pipeline(taxonomy_id, scientific_name, assembly_list, common_properties, memory, instance):
+    private_config_xml_file = common_properties["private-config-xml-file"]
+    profile = common_properties["profile"]
+    clustering_artifact = common_properties["clustering-jar-path"]
+    python = common_properties["python3-path"]
+    release_version = common_properties['release-version']
+    clustering_folder = common_properties['clustering-folder']
+    clustering_tracking_table = common_properties['clustering-release-tracker']
+
+    pipeline = LinearNextFlowPipeline()
+    for assembly in assembly_list:
+        output_directory = os.path.join(clustering_folder, f"{scientific_name.lower().replace(' ', '_')}_{taxonomy_id}", assembly)
+        os.makedirs(output_directory, exist_ok=True)
+        properties_path = create_properties_file('MONGO', None, None, assembly,
+                                                 private_config_xml_file, profile, output_directory, instance)
+        status_update_template = (f'{python} -m clustering_automation.update_clustering_status '
+                                  f'--private-config-xml-file {private_config_xml_file} '
+                                  f'--profile {profile} '
+                                  f'--clustering-tracking-table {clustering_tracking_table} '
+                                  f'--release {release_version} '
+                                  f'--assembly {assembly} '
+                                  f'--taxonomy {taxonomy_id} '
+                                  '--status {status}')  # will be filled in later
+        
+        suffix = assembly.replace('.', '_')
+        pipeline.add_process(
+            process_name=f'start_{suffix}',
+            command_to_run=status_update_template.format(status='Started'),
+        )
+        # Needed for process_directives
+        pipeline._add_new_process(NextFlowProcess(
+            process_name=f'cluster_{suffix}',
+            command_to_run=f'java -jar {clustering_artifact} --spring.config.location=file:{properties_path}',
+            process_directives={'memory': f'{memory} MB',
+                                'clusterOptions': (f'-o {output_directory}/cluster_{timestamp}.log '
+                                                   f'-e {output_directory}/cluster_{timestamp}.err')}
+        ))
+        pipeline.add_process(
+            process_name=f'end_{suffix}',
+            command_to_run=status_update_template.format(status='Completed')  # TODO: how to choose completed/failed?
```

Comment:
That would be one of the reason why it might not be such a good idea to have the status update in different nextflow steps

Reply:
I wasn't quite sure which way to go... the [release automation](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-release-automation/run_release_in_embassy/run_release_for_species.py#L56) right now does something similar to this (status update in nextflow, doesn't mark things as failed), while [remapping automation](https://github.com/apriltuesday/eva-tools/blob/master/variant-remapping-automation/remapping_automation.py#L201) does mark failures but uses a separate nextflow pipeline for each assembly.

Reply:
Yes it is something we will have to tackle when we harmonise the three automation in the same pipeline.

Reply:
I could refactor and do the status updates for each assembly outside nextflow, it will be a little silly to have a pipeline with just one step but if we do in fact add QA steps then it could make sense.

Reply:
We discussed this a bit this morning, but agreed it would make more sense to make everything consistent once we bring the automation scripts together.